### PR TITLE
Document exceptions in loggers

### PR DESCRIPTION
**Pull request by AlKun25**
## What does this PR do?
Part of #5434 

### TODO

- [x] \_\_init\_\_.py
- [x] base.py
- [x] comet.py
- [x] csv_logs.py
- [x] mlflow.py
- [x] neptune.py
- [x] tensorboard.py
- [x] test_tube.py
- [x] wandb.py

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
@AlKun25 Thank you for contribution! I think all exceptions are included in this PR which I confirmed with `grep -r "raise " pytorch_lightning/loggers/`.

File: pytorch_lightning/loggers/comet.py [hunk 1]
```diff
@@ -115,6 +115,12 @@ class CometLogger(LightningLoggerBase):
         prefix: A string to put at the beginning of metric keys.
         \**kwargs: Additional arguments like `workspace`, `log_code`, etc. used by
             :class:`CometExperiment` can be passed as keyword arguments in this logger.
+
+    Raises:
+        ImportError:
+            If required Comet package is not installed on the device
+        MisconfigurationException:
+            If neither api_key nor save_dir are passed as arguments
```
Comment (akihironitta) [inline pytorch_lightning/loggers/comet.py:123]: nit
```suggestion
            If neither ``api_key`` nor ``save_dir`` are passed as arguments
```

Comment (AlKun25) [reply pytorch_lightning/loggers/comet.py]: __Done__

File: pytorch_lightning/loggers/comet.py [hunk 2]
```diff
@@ -115,6 +115,12 @@ class CometLogger(LightningLoggerBase):
         prefix: A string to put at the beginning of metric keys.
         \**kwargs: Additional arguments like `workspace`, `log_code`, etc. used by
             :class:`CometExperiment` can be passed as keyword arguments in this logger.
+
+    Raises:
+        ImportError:
+            If required Comet package is not installed on the device
+        MisconfigurationException:
+            If neither ``api_key`` nor ``save_dir`` are passed as arguments
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/comet.py:123]:
```suggestion
            If required Comet package is not installed on the device.
        MisconfigurationException:
            If neither ``api_key`` nor ``save_dir`` are passed as arguments.
```

File: pytorch_lightning/loggers/mlflow.py [hunk 3]
```diff
@@ -79,6 +79,9 @@ def any_lightning_module_function_or_hook(self):
             Has no effect if `tracking_uri` is provided.
         prefix: A string to put at the beginning of metric keys.
 
+    Raises:
+        ImportError:
+            If required MLFlow package is not installed on the device
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/mlflow.py:84]:
```suggestion
            If required MLFlow package is not installed on the device.
```

File: pytorch_lightning/loggers/neptune.py [hunk 4]
```diff
@@ -171,6 +171,10 @@ def any_lightning_module_function_or_hook(self):
         prefix: A string to put at the beginning of metric keys.
         \**kwargs: Additional arguments like `params`, `tags`, `properties`, etc. used by
             :func:`neptune.Session.create_experiment` can be passed as keyword arguments in this logger.
+
+    Raises:
+        ImportError:
+            If required Neptune package is not installed on the device
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/neptune.py:177]:
```suggestion
            If required Neptune package is not installed on the device.
```

File: pytorch_lightning/loggers/test_tube.py [hunk 5]
```diff
@@ -74,6 +74,10 @@ def any_lightning_module_function_or_hook(self):
             the user has defined the `self.example_input_array` attribute in their
             model.
         prefix: A string to put at the beginning of metric keys.
+
+    Raises:
+        ImportError:
+            If reqired TestTube package is not installed on the device
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/test_tube.py:80]:
```suggestion
            If required TestTube package is not installed on the device.
```

File: pytorch_lightning/loggers/wandb.py [hunk 6]
```diff
@@ -61,6 +61,12 @@ class WandbLogger(LightningLoggerBase):
         \**kwargs: Additional arguments like `entity`, `group`, `tags`, etc. used by
             :func:`wandb.init` can be passed as keyword arguments in this logger.
 
+    Raises:
+        ImportError:
+            If required WandB package is not installed on the device
+        MisconfigurationException:
+            If model checkpoints are uploaded in offline mode
+
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/wandb.py:69]:
```suggestion
            If required WandB package is not installed on the device.
        MisconfigurationException:
            If both ``log_model`` and ``offline``is set to ``True``.

```

**Review by rohitgr7 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**